addons/google_gmail/models/ir_mail_server.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+
+from odoo import models, api
+
+
+class IrMailServer(models.Model):
+    """Represents an SMTP server, able to send outgoing emails, with SSL and TLS capabilities."""
+
+    _name = 'ir.mail_server'
+    _inherit = ['ir.mail_server', 'google.gmail.mixin']
+
+    @api.onchange('smtp_encryption')
+    def _onchange_encryption(self):
+        if not self.use_google_gmail_service:
+            super(IrMailServer, self)._onchange_encryption()
+
+    @api.onchange('use_google_gmail_service')
+    def _onchange_use_google_gmail_service(self):
+        if self.use_google_gmail_service:
+            self.smtp_host = 'smtp.gmail.com'
+            self.smtp_encryption = 'starttls'
+            self.smtp_port = 587
+        else:
+            self.smtp_encryption = 'none'
+            self.smtp_pass = ''
+            self.smtp_host = ''
+            self.google_gmail_authorization_code = False
+            self.google_gmail_refresh_token = False
+
+    def _smtp_login(self, connection, smtp_user, smtp_password):
+        if len(self) == 1 and self.use_google_gmail_service:
```

Comment:
Bit weird, due to the way we do the smtp login ... Note that with a void recordset, self.use_google_gmail_server will be Falsy. We could just say "if self.use_google_gmail_service" and this implies an ensure one + a valid self. Your approach is more explicit, as you prefer.

Reply:
I prefer to keep the `len(self) == 1`, as you said, it's a bit more explicit